pkg/virtctl/console/console.go

```diff
@@ -93,19 +93,60 @@ func (c *Console) Run(cmd *cobra.Command, args []string) error {
 	writeStop := make(chan error)
 	readStop := make(chan error)
 
+	// Wait until the virtual machine is in running phase, user interrupt or timeout
+	runningChan := make(chan error)
+	timeoutChan := time.NewTicker(time.Duration(timeout) * time.Minute)
+	waitInterrupt := make(chan os.Signal, 1)
+	signal.Notify(waitInterrupt, os.Interrupt)
+
 	go func() {
 		con, err := virtCli.VirtualMachineInstance(namespace).SerialConsole(vmi)
-		if err != nil {
-			resChan <- err
-			return
+		for err != nil {
```

Comment:
Sorry for the tardy review. I thought I had reviewed yesterday but my comment seems to have disappeared.

It might be worth looking into the "k8s.io/apimachinery/pkg/util/wait" library. There's a PollImmediate function that already handles the busywork of polling on an interval with a timeout/error conditions. Using that could possibly simplify this code a bit.

Reply:
I take a look into the function and make the implementation but this not simplify the code.
This help with the timeout but not with the user interrupt.
Also we need to create a new goroutine after the PollImmediate success to create a new connection to the console service for the connection stream.